src/main/scala/io/circe/bson/BsonCodecInstances.scala

```diff
@@ -55,7 +55,11 @@ trait BsonCodecInstances {
     case BSONJavaScriptWS(value) => Right(Json.fromString(value))
     case BSONMaxKey              => Left(readerFailure(bson))
     case BSONMinKey              => Left(readerFailure(bson))
-    case BSONObjectID(_)         => Left(readerFailure(bson))
+    case id: BSONObjectID        =>
+      BSONObjectID.parse(id.stringify) match {
```

Comment:
I'm still not sure I understand under what circumstances a `BSONObjectID` would have a `id.stringify` member that couldn't be parsed by `BSONObjectID.parse`?

Reply:
You could also just write:
```scala
case id: BSONObjectID        => Right(Json.fromString(id.stringify))
``` , I guess. It is only an additional round-tripping safety check since the validity of a BSONObjectID is evaluated by conversions which are not totally type-safe in my opinion.
If we'd say we would aim for performance, we could also leave it up to the reactivemongo library to guarantee the soundness of the conversion.

Reply:
I just tested the version without the round trip: 
`sbt test` produces:

```
[info] - BsonCodecInstances should round-trip JSON values *** FAILED ***
[info]   TestFailedException was thrown during property evaluation.
[info]     Message: Right({
[info]     "驯稟ྶ뷺ὃᱺ㌧ꜟψ贷疅釤厧䶾驈퓻嚥視" : -33614283950904051868112239385e-4999
[info]   }) did not equal Right({
[info]     "驯稟ྶ뷺ὃᱺ㌧ꜟψ贷疅釤厧䶾驈퓻嚥視" : -0.0
[info]   })
[info]     Location: (BsonCodecInstancesTest.scala:23)
[info]     Occurred when passed generated values (
[info]       arg0 = {
[info]     "驯稟ྶ뷺ὃᱺ㌧ꜟψ贷疅釤厧䶾驈퓻嚥視" : -33614283950904051868112239385e-4999
[info]   } // 8 shrinks
[info]     )
```